### PR TITLE
Fix tab audio muting button

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/21829

Chromium used to support its own tab muting button but that was removed in 2018. Since then, we added our own button and a flag (defaulting to enabled) to control it. In CR100, Chromium reimplemented tab muting via https://chromium-review.googlesource.com/c/chromium/src/+/3123736.

In order to account for this, I removed our implementation of the tab muting button and our Brave-specific flag. We now use Chromium's implementation of the button and their flag (brave://flags/#enable-tab-audio-muting).

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Clean profile
- Visit https://www.youtube.com/watch?v=NSZ-D8YkodI
- Play the video
- Verify that pressing the audio mute icon mutes the video
- Verify that pressing the audio mute icon again unmutes the video
- Visit brave://flags/#enable-tab-audio-muting
- Set feature to disabled
- Relaunch browser
- Visit https://www.youtube.com/watch?v=NSZ-D8YkodI
- Play the video
- Verify that audio mute icon isn't interactive

### CODE REVIEW

**Review by bsclifton — APPROVED** [review]
LGTM!
- removes our brave://flags entry for this brave://flags/#tab-audio-icon-interactive
- defaults the Chromium one to TRUE (using chromium_src override for setting default, so user can override with brave://flags/#enable-tab-audio-muting
- removes our custom code handling muting/unmuting

I think this will have an impact to the way it's rendered. @nullhook did some changes with contrast. Would be good to have him check out 😄 See https://github.com/brave/brave-browser/issues/17102 for more info

**Comment by bsclifton** [comment]
Note: we could add our own UI element to brave://settings and tie to brave://flags/#enable-tab-audio-muting

See https://github.com/brave/brave-browser/issues/6690

**Review by simonhong — APPROVED** [review]

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "brave/browser/ui/views/tabs/brave_alert_indicator_button.h"
-
-// Set alert indicator's pos to start of the title and
-// move title after the alert indicator.
-// Title right should respect close btn's space
-#define BRAVE_UI_VIEWS_TABS_TAB_ALERT_INDICATOR_POSITION           \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/views/tabs/tab.cc:11]: Oh, wait. I think this is not related with this PR?

Comment (simonhong) [reply chromium_src/chrome/browser/ui/views/tabs/tab.cc:11]: This change came from https://github.com/brave/brave-core/pull/9445 and it's for re-positioning alert button

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
-  alert_indicator_button_->SetX(title_left - after_title_padding); \
-  title_right = close_x - after_title_padding;                     \
-  if (showing_close_button_)                                       \
-    title_right = close_x - after_title_padding;                   \
-  title_left = alert_indicator_button_->x() +                      \
-               alert_indicator_button_->width() + after_title_padding;
-
-#define BRAVE_UI_VIEWS_TABS_TAB_UPDATE_ICON_VISIBILITY \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/views/tabs/tab.cc:19]: This one also.
I think we should only delete `AlertIndicatorButton` re-define?

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc (hunk 1 above)
Comment (emerick) [reply chromium_src/chrome/browser/ui/views/tabs/tab.cc:11]: I see, I'll fix that. Thanks for catching!

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc (hunk 2 above)
Comment (emerick) [reply chromium_src/chrome/browser/ui/views/tabs/tab.cc:19]: Agreed, I'll fix that - thanks!

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc (hunk 1 above)
Comment (emerick) [reply chromium_src/chrome/browser/ui/views/tabs/tab.cc]: Fixed.

File: chromium_src/chrome/browser/ui/views/tabs/tab.cc (hunk 2 above)
Comment (emerick) [reply chromium_src/chrome/browser/ui/views/tabs/tab.cc:19]: Fixed.

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: chromium_src/media/base/media_switches.cc [hunk 3]
```diff
@@ -11,6 +11,7 @@ namespace media {
 
 OVERRIDE_FEATURE_DEFAULT_STATES({{
     {kLiveCaption, base::FEATURE_DISABLED_BY_DEFAULT},
+    {kEnableTabMuting, base::FEATURE_ENABLED_BY_DEFAULT},
```
Comment (mkarolin) [inline chromium_src/media/base/media_switches.cc:14]: We probably want to add the feature to `BraveMainDelegateBrowserTest.EnabledFeatures` in `brave/app/brave_main_delegate_browsertest.cc`

**Review by mkarolin — APPROVED** [review]
LGTM + update browser test

File: chromium_src/media/base/media_switches.cc (hunk 3 above)
Comment (emerick) [reply chromium_src/media/base/media_switches.cc:14]: Good point, updated that browser test.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.39.5 Chromium: 100.0.4896.46 (Official Build) nightly (64-bit)
-- | --
Revision | 5ca33821b2211805855c77d334353d27c616a7ca-refs/branch-heads/4896@{#584}
OS | Windows 11 Version 21H2 (Build 22000.556)
```

Two points that were already raised and @brave/user-support-team was made aware as well:

* Because we're removing/deprecating `brave://flags/#tab-audio-icon-interactive` in favour of `brave://flags/#enable-tab-audio-muting`, any user that disabled the ability to mute the tabs via the speaker icon will need to re-enable it via `brave://flags/#enable-tab-audio-muting`.
* Seems like the behaviour has changed in `C100` were you need to be on the active tab to be able to click on the speaker. I guess this prevents accidentally clicking on a speaker on another tab which was an issue before.

Went through the STR/Cases outlined via https://github.com/brave/brave-core/pull/12719#issue-1178350366 and ensured the following:

* verified that `brave://flags/#tab-audio-icon-interactive` was removed
* verified that `brave://flags/#enable-tab-audio-muting` is visible/available via `brave://flags`
* ensured that `Default` for `brave://flags/#enable-tab-audio-muting` means being able to mute the tab via the speaker icon
* ensured that clicking on the speaker icon mutes the tab (as per the above, it needs to be the active tab)
* ensured that `Mute Site` works when `brave://flags/#enable-tab-audio-muting` is enabled/disabled (right click tabs)
* ensured that you can't click/mute the tab via the speaker icon when disabling `brave://flags/#enable-tab-audio-muting`

As mentioned above, if upgrading from`C99` and ``brave://flags/#tab-audio-icon-interactive` is disabled, the user will need to re-disable it via `brave://flags/#enable-tab-audio-muting`.
